ext/flash/sendfile.rs

```diff
@@ -21,9 +83,50 @@ impl SendFile {
       let count = 0x7ffff000;
       let mut offset = self.written as libc::off_t;
 
+      // sendfile() with TCP_CORK
+      if self.sending_headers {
+        let opt = 1;
+        // SAFETY: call to libc
+        let length = unsafe {
+          libc::setsockopt(
+            self.io.1,
+            libc::SOL_SOCKET,
+            libc::TCP_CORK,
+            &opt as *const _ as _,
+            4,
+          );
+          libc::writev(
+            self.io.1,
+            self.slices.as_ptr() as _,
+            self.slices.len() as i32,
+          )
+        };
+
+        let io_vec = &mut self.slices;
+        let accumulated_len = advance_io_vec(io_vec, length as usize);
+        if io_vec.is_empty() {
+          self.sending_headers = false;
+          self.written += length as usize - accumulated_len;
+        }
+        return Ok(length as usize);
```

Comment:
I don't quite understand this return statement. Don't you want to move on to the actual sendfile call now?